modules/logger/src/LogStep.ts

```diff
@@ -135,17 +138,34 @@ export class LogStep {
 	 *
 	 * @internal
 	 */
-	activate(enableWrite = this.#stream === process.stderr && !process.stderr.isTTY) {
-		if (this.#active) {
+	activate(enableWrite = !('isTTY' in this.#stream && this.#stream.isTTY)) {
+		if (this.#active && this.#writing === enableWrite) {
 			return;
 		}
 
 		this.#active = true;
+
 		if (enableWrite) {
 			this.#enableWrite();
 		}
 	}
 
+	/**
+	 * @internal
+	 */
+	deactivate() {
+		if (!this.#active) {
+			return;
+		}
+
+		this.#active = false;
+		if (this.#writing) {
+			this.#buffer.off('data', noop);
```

Comment:
I meant to ask about this earlier what is this doing? looks like its unregistering a noop listener?

Reply:
Yeah, that's added on line 176... it prevents needing to manually call `this.#buffer.read()` to get stuff: https://nodejs.org/api/stream.html#two-reading-modes It's a bit wonky.

Reply:
eldritch streaming knowledge got it